lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4311,34 +4311,10 @@ FIRCircuitParser::parsePortList(SmallVectorImpl<PortInfo> &resultPorts,
     resultPortLocs.push_back(info.getFIRLoc());
   }
 
-  // Helper for the temporary check rejecting input-oriented refs.
-  std::function<bool(Type, bool)> hasInputRef = [&](Type type,
-                                                    bool output) -> bool {
-    auto ftype = type_dyn_cast<FIRRTLType>(type);
-    if (!ftype || !ftype.containsReference())
-      return false;
-    return TypeSwitch<FIRRTLType, bool>(ftype)
-        .Case<RefType>([&](auto reftype) { return !output; })
-        .Case<OpenVectorType>([&](OpenVectorType ovt) {
-          return hasInputRef(ovt.getElementType(), output);
-        })
-        .Case<OpenBundleType>([&](OpenBundleType obt) {
-          for (auto field : obt.getElements())
-            if (hasInputRef(field.type, field.isFlip ^ output))
-              return true;
-          return false;
-        });
-  };
-
   // Check for port name collisions.
   SmallDenseMap<Attribute, SMLoc> portIds;
   for (auto portAndLoc : llvm::zip(resultPorts, resultPortLocs)) {
     PortInfo &port = std::get<0>(portAndLoc);
-    // See #4812 and look through the reference input test collection
-    // and ensure they work before allowing them from user input.
-    if (hasInputRef(port.type, port.isOutput()))
-      return emitError(std::get<1>(portAndLoc),
-                       "input probes not yet supported");
```

Comment:
Locally, some time ago, added verifier that rejected input probes on public modules.  Other than test churn, I'm not sure why that wasn't landed.  Should go revive, suspect it was just unreachable and so not worth the change.